### PR TITLE
Plan 028: knowledge dispatch + parse_alink reconciliation

### PR DESCRIPTION
**Pull request by gitronald**
## Summary

Implements the dispatch/reconciliation core of **Plan 028** (knowledge parsers rethink + `parse_alink` reconciliation), plus bookkeeping for the merged Plan 027. Built on the `feature/v0.9.0` line.

All phases are behavior-preserving: **296 tests pass (+18 new), 66 snapshots unchanged, ruff clean.**

### Phases

- **Phase 1 — `parse_alink` reconciliation.** Replaced four near-identical private `parse_alink` copies (`general`, `knowledge`, `knowledge_rhs`, `top_image_carousel`) with one parameterized helper in a new `component_parsers/_common.py`:
  `parse_alink(a, sep="", data_url_fallback=False)`. Missing `href` now yields `url=None` (lenient) rather than raising; every current call site already guards href presence, so output is unchanged. Shared `parse_alink_list` moved off `general.py`.

- **Phase 2 — table-driven knowledge dispatch.** Converted `parse_knowledge_panel`'s 13-branch `if/elif` cascade into an ordered detect-and-handle table (`_SUBTYPE_HANDLERS` + `_subtype_panel` fallback), mirroring `classifiers/main.py`. Behavior-preserving by construction, including the two conditional-consumer branches (`things_to_know` claims the panel even when the heading text is unrecognized; the dynamic `JNkvid` slug branch falls through to `panel` when the level-2 heading is absent). Adds `tests/test_knowledge_dispatch.py` (18 pins) covering the five sub_types no SERP fixture exercised + the two edge cases.

- **Phase 3a — registry close-out.** Added `panel_rhs` to the `knowledge` `ComponentType` sub_types and documented the open dynamic-slug space.

### Plan bookkeeping

- Marked Plan 027 `completed` (its PR #139 was merged).
- Marked Plan 028 `completed` for this scope; split the deferred 5th open question — `details`-schema typed alignment — into **Plan 029**, since it needs a concrete target schema defined first and changes output broadly (best reviewed in isolation).

### Resolved design questions (4 of 5)

| Question | Resolution |
|---|---|
| Dispatch shape | Table-driven |
| `knowledge` vs `knowledge_rhs` sharing | Stay separate; only `parse_alink` was true duplication (now unified) |
| Dynamic slug policy | Kept open, documented |
| Link-helper location | `component_parsers/_common.py` |
| `details` schema alignment | Deferred → Plan 029 |

https://claude.ai/code/session_01XQ6dz41mc3okJAbfFRpniF

---
_Generated by [Claude Code](https://claude.ai/code/session_01XQ6dz41mc3okJAbfFRpniF)_

### CODE REVIEW

File: WebSearcher/component_parsers/_common.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    A missing ``href`` (and ``data-url``, when ``data_url_fallback``) yields
+    ``url=None`` rather than raising -- callers needing strict behavior must
+    guard ``"href" in a.attributes`` before calling.
+    """
+    if data_url_fallback:
+        url = a.attributes.get("href") or a.attributes.get("data-url")
+    else:
+        url = a.attributes.get("href")
+    return {"url": url, "text": get_text(a, sep) or ""}
```
[the remark this answers is not in the document]
Comment (gitronald) [reply WebSearcher/component_parsers/_common.py:33]: This is the intended behavior per a deliberate decision on this PR: `parse_alink` was unified on **lenient** semantics (missing url → `None`) across all four call sites, rather than the old per-site mix of strict `KeyError` and `""`.

For the carousel specifically, the call site guards before calling:
```python
if "href" in a.attributes or "data-url" in a.attributes:
    items.append(parse_alink(a, "|", data_url_fallback=True))
```
so `url` is the `href` or `data-url` value; `None` is only reachable if a present attribute is an empty string, which doesn't occur in practice (no snapshot/test moved). Keeping it uniform avoids the carousel being the one site that coalesces to `""` while every other lenient site yields `None`. Leaving open in case you'd prefer the old `""` coalescing here.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (gitronald) [reply WebSearcher/component_parsers/_common.py:33]: Confirmed as the intended behavior — keeping the lenient `url=None` here for uniformity with the other call sites. Decision recorded in the plan doc (`docs/plans/028-knowledge-parsers-and-alink-reconciliation.md`, "Decisions (locked)") in `d4bcd32`. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_
